x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPlugin.java

```diff
@@ -121,14 +123,28 @@ public class EnrichPlugin extends Plugin implements SystemIndexPlugin, IngestPlu
         return String.valueOf(maxConcurrentRequests * maxLookupsPerRequest);
     }, val -> Setting.parseInt(val, 1, Integer.MAX_VALUE, QUEUE_CAPACITY_SETTING_NAME), Setting.Property.NodeScope);
 
-    public static final Setting<Long> CACHE_SIZE = Setting.longSetting("enrich.cache_size", 1000, 0, Setting.Property.NodeScope);
+    public static final Setting<Long> CACHE_SIZE = new Setting<>("enrich.cache_size", (String) null, (String s) -> {
+        if (s == null) {
+            return null;
+        }
+        return Setting.parseLong(s, 0, "enrich.cache_size");
+    }, Setting.Property.NodeScope);
```

Comment:
For now, I didn't change the setting to also allow a `ByteSizeValue` or a percentage (of max heap space). As we discussed internally, there are probably not many users manually configuring this setting, so it might not be worth putting in effort into this setting. Any other thoughts/opinions are welcome.

Reply:
Hm this approach doesn't work like I thought it would. I'll wait with updating this before someone has had a chance to respond to my comment above, as that would require a change here as well.

Reply:
@dakrone @joegallo do you have any thoughts on whether/how we should allow users to configure the enrich cache size? Obviously, we'll still have the flat document count option for BWC. On the one hand, it feels off not to allow users to configure the enrich cache in terms of bytes or a percentage when the default is based on a percentage. But OTOH, I'm not sure if the effort and added complexity is worth it, given that there are probably not many users out there even configuring this. The "effort and added complexity" I'm referring to is about having a single setting that takes three units (flat document count, bytes, and percentage).

Reply:
I think we should allow specifying it in bytes or percentages, the same way we have other caches configured. Then we can (in a separate PR) at least start the deprecation train for the document-count configuration. We should also then change the default to be percentage-based rather than document-based.